ls/src/output.rs

```diff
@@ -0,0 +1,283 @@
+use std::{
+    io::{self, BufWriter, Write},
+    os::unix::fs::MetadataExt,
+};
+
+use coreutils_core::{
+    os::tty::{is_tty, tty_dimensions},
+    BString, ByteSlice,
+};
+
+use term_grid::{Alignment, Cell, Direction, Filling, Grid, GridOptions};
+
+extern crate chrono;
+
+use crate::{
+    file::{FileColor, Files},
+    flags::Flags,
+    table::{Row, Table},
+};
+
+/// Outputs in the provided files in a style depending on the flags provided.
+pub(crate) fn output(result: Files, flags: Flags) -> i32 {
+    let mut exit_code = 0;
+
+    let mut writer = BufWriter::new(io::stdout());
+
+    if flags.show_list() {
+        match list(result, &mut writer, flags) {
+            Ok(_) => {},
+            Err(err) => {
+                eprintln!("ls: {}", err);
+                exit_code = 1;
+            },
+        }
+    } else if flags.show_grid() {
+        let direction = if flags.order_left_to_right && !flags.order_top_to_bottom {
+            Direction::LeftToRight
+        } else {
+            Direction::TopToBottom
+        };
+
+        match grid(result, &mut writer, direction) {
+            Ok(_) => {},
+            Err(err) => {
+                eprintln!("ls: {}", err);
+                exit_code = 1;
+            },
+        }
+    } else {
+        match default(result, &mut writer, flags) {
+            Ok(_) => {},
+            Err(err) => {
+                eprintln!("ls: {}", err);
+                exit_code = 1;
+            },
+        }
+    }
+
+    exit_code
+}
+
+/// Writes the provided files in the default format.
+pub(crate) fn default<W: Write>(files: Files, writer: &mut W, flags: Flags) -> io::Result<()> {
+    if !is_tty(&io::stdout()) {
```

Comment:
Are we checking for the right thing here? In my mind I was thinking that here we wanted to check if the underlying `W` in `BufWriter<W>` is not a TTY, not specifically the stdout.

Let me know what you think, I may have misunderstood this part of the code

Reply:
I may be misunderstanding, but I believe it would be the same either way since the `BufWriter` instance is loaded with the `stdout` file descriptor.I went ahead and made this change though.  I had to refactor to writer parameters to use `BufWriter<Stdout>` as it's type in order to get the file descriptor reference. Although I cannot remember why I was using the parameterized type for the writer to begin with.